### PR TITLE
Move tests from okhttp-tests/src/test/java to okhttp/src/test/java

### PR DESCRIPTION
**Pull request by swankjesse**
The previous module structure was necessary because Maven wouldn't allow
mockwebserver to depend on okhttp/src and for okhttp/test to depend on
mockwebserver. With Gradle this constraint is lifted and we can fold
everything into a single module.

I'm in a rush to do this now because it's necessary for Kotlin 'internal'
tests to be in the same module.

### CODE REVIEW

File: okhttp/build.gradle [hunk 1]
```diff
@@ -29,8 +29,16 @@ dependencies {
   compileOnly deps.jsr305
   compileOnly deps.animalSniffer
 
+  testImplementation deps.okio
+  testImplementation project(':okhttp-testing-support')
+  testImplementation project(':okhttp-tls')
+  testImplementation project(':okhttp-urlconnection')
```
Comment (JakeWharton) [inline okhttp/build.gradle]: This one feels strange...

File: okhttp/build.gradle [hunk 2]
```diff
@@ -29,8 +29,16 @@ dependencies {
...
+  testImplementation project(':mockwebserver')
+  testImplementation project(':okhttp-logging-interceptor')
```
Comment (JakeWharton) [inline okhttp/build.gradle]: And this one...

Comment (swankjesse) [reply okhttp/build.gradle]: EventListenerTest.java uses it. Can fix, but not this PR.

File: okhttp/build.gradle (hunk 1 above)
Comment (swankjesse) [reply okhttp/build.gradle]: CookiesTest.java uses JavaNetCookieJar. Can fix, but not this PR.

**Pull request merged**